src/polkit/polkitunixprocess.c

```diff
@@ -1131,6 +1361,437 @@ get_start_time_for_pid (pid_t    pid,
   return start_time;
 }
 
+static guint64
+get_cgroupid_for_pidfd (gint    pidfd,
+                        GError **error)
+{
+#ifdef __linux__
+  guint64 cgroupid;
+  gint pid;
+  gchar *filename;
+  gchar *cgroup_path;
+  gchar *contents;
+  size_t length;
+  gchar *p;
+  struct file_handle *fhp;
+  int mount_id;
+
+  pid = 0;
+  filename = NULL;
+  contents = NULL;
+  cgroup_path = NULL;
+  cgroupid = 0;
+  fhp = NULL;
+
+  g_return_val_if_fail (pidfd >= 0, 0);
+
+  pid = polkit_unix_process_get_pid_from_pidfd (pidfd, error);
+  if (pid <= 0)
+    return 0;
+
+  filename = g_strdup_printf ("/proc/%d/cgroup", pid);
+  if (!filename)
+    {
+      g_set_error (error,
+                   POLKIT_ERROR,
+                   POLKIT_ERROR_FAILED,
+                   "Error allocating memory for procfs path");
+      goto out;
+    }
+
+  if (!g_file_get_contents (filename, &contents, &length, error))
+    goto out;
+
+  /* The format is '0::/cgroup/path\n */
+  p = strchr (contents, ':');
+  if (p == NULL)
+    {
+      g_set_error (error,
+                   POLKIT_ERROR,
+                   POLKIT_ERROR_FAILED,
+                   "Error parsing file %s",
+                   filename);
+      goto out;
+    }
+  p += 2; /* skip '::' */
+  if (p - contents >= (int) length)
+    {
+      g_set_error (error,
+                   POLKIT_ERROR,
+                   POLKIT_ERROR_FAILED,
+                   "Error parsing file %s",
+                   filename);
+      goto out;
+    }
+
+  contents[length - 1] = '\0'; /* remove the newline */
+
+  cgroup_path = g_strdup_printf ("/sys/fs/cgroup%s", p);
+  if (!cgroup_path)
```

Comment:
glib aborts on oom anyway, you don't need to check for oom hence. not here not anywhere else